synapse/lib/remcycle.py

```diff
@@ -0,0 +1,846 @@
+                    "ingest_definition": {
+                      "ingest": {
+                        "forms": [
+                          [
+                            "inet:ipv4",
+                            {
+                              "var": "ip"
+                            }
+                          ]
+                        ],
+                        "vars": [
+                          [
+                            "ip",
+                            {
+                              "path": "ip"
+                            }
+                          ]
+                        ]
+                      }
+                    }
+                  },
+                  "url": "http://vertex.link/api/v4/geoloc/{{someplace}}/info?domore={{domore}}&apikey={APIKEY}",
+                  "vars": [
+                    ["APIKEY", "8675309"]
+                  ]
+                },
+                "https": {
+                  "doc": "Get the vertex project landing page.",
+                  "http": {
+                    "validate_cert": false
+                  },
+                  "url": "https://vertex.link/"
+                }
+              },
+              "doc": "Grab Vertex.link stuff",
+              "http": {
+                "user_agent": "Totally Not a Python application."
+              },
+              "namespace": "vertexproject"
+            }
+
+        ::
+
+        The following keys are required:
+
+            * namespace: String identifier for all APIs present in the
+              configuration.  Must be locally unique.
+            * doc: Simple string describing the overall namespace.
+            * apis: Dictionary containing configuration values for API
+              endpoints. See Nyx object for details of how this data should be
+              shaped.  The keys of this dictionary, when joined with the
+              namespace of the configuration, form the name of the APIs for
+              later use.  Given the example above, the following APIs would
+              be registered:
+                - vertexproject:geoloc
+                - vertexproject:https
+
+        The following keys are optional:
+
+            * http: Global HTTP Request arguments which will be the basis
+              for creating HTTPRequest objects.
+
+
+        :param config: Dictionary containing the configuration information.
+        :param reload_config: If true and the namespace is already registered,
+                              the existing namespace will be removed and the new
+                              config added.
+        :return: None
+        :raises: NameError if the existing namespace is registered or a
+                 invalid HTTP value is provided.
+        :raises: Other exceptions are possible if the configuration isn't
+                 shaped properly, this would likely come from duck typing.
+        '''
+        try:
+            self._parseConfig(config=config, reload_config=reload_config)
```

Comment:
remove keyword param syntax for mandatory params for style consistency. :D  ( here and elsewhere )